### ISSUE TITLE
Performance difference between Xgboost 0.81 and 0.60 on windows

### ISSUE THREAD

**Issue by csaiedu** [dmlc/xgboost#3903]
With Xboost 0.81 on windows 10 (gpu enabled), nthread is set to None implying 1, which require to set 
that default to -1 if training on the cpu rather than the gpu. This creates portability issues.

When training with the code below:
With nthread=-1 the training time takes 12 seconds, while with my older xgboost install (xgboost 0.6 non gpu) the time is half at 6 seconds. Are there other significant default parameters difference beside the nthread to set manually to match the performance ? Or are there fundamental differences in the gpu build of xgboost 0.81 that makes it slower when training in cpu only mode?

For completeness with gpu_exact it only takes 4 seconds on xgboost 0.81.

```
from sklearn import datasets  
from sklearn.model_selection import train_test_split  
from sklearn.model_selection import cross_val_score


X, y = datasets.make_classification(n_samples=100000, n_features=20,  

                                    n_informative=2, n_redundant=10,

                                    random_state=42)

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3,  

                                                    random_state=42)



from xgboost.sklearn import XGBClassifier  

from xgboost.sklearn import XGBRegressor



xclas = XGBClassifier()  # and for classifier  

xclas.fit(X_train, y_train)  

xclas.predict(X_test)  


print(cross_val_score(xclas, X_train, y_train)  )

```
xgboost 0.81 parameters default

XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,
       colsample_bytree=1, gamma=0, learning_rate=0.1, max_delta_step=0,
       max_depth=3, min_child_weight=1, missing=None, n_estimators=100,
       n_jobs=1, nthread=None, objective='binary:logistic', random_state=0,
       reg_alpha=0, reg_lambda=1, scale_pos_weight=1, seed=None,
       silent=True, subsample=1)

xgboost 0.60 parameters default

XGBClassifier(base_score=0.5, colsample_bylevel=1, colsample_bytree=1,
       gamma=0, learning_rate=0.1, max_delta_step=0, max_depth=3,
       min_child_weight=1, missing=None, n_estimators=100, nthread=-1,
       objective='binary:logistic', reg_alpha=0, reg_lambda=1,
       scale_pos_weight=1, seed=0, silent=True, subsample=1)

**Comment by hcho3** [comment]
Did you compile both versions from source? The binary wheel is compiled using Visual Studio 2015 + Windows Server 2008 (Vista) combination, so performance-wise may not be optimal.

**Comment by trivialfis** [comment]
Took a quick glance, I don't think the GPU has much contribution to the run time since I barely saw any activity on device monitor.  Will do some digging in the future.

**Comment by csaiedu** [comment]
I have used the binary wheels so may be it explains the drop in performance. It might need to wait for an anaconda release of 0.81 then.

**Comment by hcho3** [comment]
@csaiedu You should compile from the source for best results. We don't manage Anaconda distribution, so I don't have much say about its distribution cycle.
